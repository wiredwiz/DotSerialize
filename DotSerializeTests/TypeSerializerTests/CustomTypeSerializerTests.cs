using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Org.Edgerunner.DotSerialize.Tests.DataTypes;
using Org.Edgerunner.DotSerialize.Tests.TypeSerializers;
// ReSharper disable InconsistentNaming

namespace Org.Edgerunner.DotSerialize.Tests.TypeSerializerTests
{
   [TestClass]
   public class CustomTypeSerializerTests
   {
      private const string SerializeDogResultsInExpectedOutput_Approved =
         "CustomTypeSerializerTests.SerializeDogResultsInExpectedOutput.approved.xml";

      [TestCleanup]
      private void CleanUp()
      {
         Utilities.DeleteFile(SerializeDogResultsInExpectedOutput_Approved);
      }

      [TestMethod]
      public void SerializeDogResultsInExpectedOutput()
      {
         var dog = new Dog("Fido", "Golden Retriever", true, null)
         {
            BirthDate = new DateTime(2003, 5, 10),
            Collar = DogCollarFactory.GetCollar(20, true)
         };
         var serializer = new Serializer();
         serializer.RegisterType<Dog>().ToTypeSerializer<DogSerializer>().WithConstructorArgument("title", "King");
         string xml = serializer.SerializeObject(dog);
         Approvals.VerifyXml(xml);
      }

      [TestInitialize]
      private void Setup()
      {
         Utilities.ExtractEmbeddedFile(SerializeDogResultsInExpectedOutput_Approved);
      }
   }
}
