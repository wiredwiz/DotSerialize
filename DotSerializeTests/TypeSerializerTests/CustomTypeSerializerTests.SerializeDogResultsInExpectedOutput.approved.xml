<Dog xmlns:dts="http://www.dotserialize.com/Schema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.Dog,  DotSerializeTests" dts:id="1" dts:src="true">
  <Name>King Fido</Name>
  <BirthDate>2003-05-10T00:00:00.0000000</BirthDate>
  <Breed>Golden Retriever</Breed>
  <Collar dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.DogCollar,  DotSerializeTests" dts:id="2" dts:src="true">
    <Length>20</Length>
    <Studded>True</Studded>
  </Collar>
  <_Owner xsi:nil="true" />
  <HasCollar>True</HasCollar>
  <Nose>Normal</Nose>
  <Bone dts:type="Org.Edgerunner.DotSerialize.Tests.DataTypes.Bone,  DotSerializeTests" Type="Chicken" Width="0" Diameter="0" />
  <Weight>0</Weight>
  <Height>0</Height>
</Dog>